src/render/resolved_dependencies.rs

```diff
@@ -553,7 +553,16 @@ pub async fn resolve_dependencies(
         }
     }
 
-    let match_specs = specs.iter().map(|s| s.spec().clone()).collect::<Vec<_>>();
+    let match_specs = specs
```

Comment:
this looks fine but should only happen when the `merge_build_host` is true. 
Also in that case, the build env does not need to be created.